In [1]:
import pandas as pd
from selenium import webdriver 
from bs4 import BeautifulSoup
import time
import re

In [2]:
from datetime import date, timedelta, datetime
import os.path

In [3]:
t0 =  date.today()

In [4]:
news_url='http://www.517huizhou.com/forum.php?mod=forumdisplay&fid=135'

datetime_str = t0
fname1 = './data/general/temp/' + '517huizhou-huiwenhua' + '.csv'
fname2 = './data/general/' + '517huizhou-huiwenhua' + '.csv'


In [5]:
def get_columns():
    return ('dt', 'source', 'domain', 'url', 'title', 'summary', 'image', 'brief', 'body', 'raw_page', 'from') 
def get_empty_columns():
    return ('', '', '', '', '', '', '', '', '', '', '') 
def get_empty_columns1():
    return ('', '', '', '', '', '', '', '', '', '', '','') 



In [6]:
browser = webdriver.Firefox() 
#browser = webdriver.Chrome('/Users/nhu2000/Downloads/Chromedriver')

In [7]:
if os.path.isfile(fname1): 
    df = pd.read_csv(fname1)
    idx = len(df.index)
else:
    df = pd.DataFrame(columns = get_columns())
    idx = 0

In [8]:
if os.path.isfile(fname2): 
    df_check = pd.read_csv(fname2)
    idx = len(df.index)
else:
    df_check = pd.DataFrame(columns = get_columns())
    idx = 0

if os.path.isfile(fname2): 
    df_0 = pd.read_csv(fname2)
else:
    df_0 = pd.DataFrame(columns = get_columns())

In [9]:
print len(df_check)

0


In [10]:
print len(df)

0


browser.get(news_url)
time.sleep(5)

In [7]:
url_0 = news_url
print url_0
browser.get(url_0)
time.sleep(5)

http://www.517huizhou.com/forum.php?mod=forumdisplay&fid=135


In [24]:
not_existed = True
df = pd.DataFrame(columns = get_columns())
idx = 0

In [25]:
soup = BeautifulSoup(browser.page_source)
sub_text = soup.findAll('tbody', {'id': lambda L: L and L.startswith('normalthread')})

#print sub_text

for item in sub_text:
    #print 'www.hzwh.com/' + item.a['href']
    #print t1.text
    #print item.text
    t1 = item.find('a', {'class': 's xst'})
    t2 = item.find('td', {'class': 'by'})
    if t1 != None:
        t3 = 'http://www.517huizhou.com/' + t1['href']

    if not_existed == True and t1 != None:   
        print 'add --------'
        print t1.text
        print t3
        print t2.em.text
        df.loc[idx] = get_empty_columns()        
        df.loc[idx]['url'] = t3
        df.loc[idx]['title'] = t1.text
        df.loc[idx]['source'] = '徽州大论坛'
        df.loc[idx]['dt'] = t2.em.text
        idx += 1

add --------
这次有题目，谁第一个答对赠书一本。看图说徽州系列讲座第十一讲
http://www.517huizhou.com/forum.php?mod=viewthread&tid=90595&extra=page%3D1
2016-4-6
add --------
听老土讲徽州
http://www.517huizhou.com/forum.php?mod=viewthread&tid=90951&extra=page%3D1
2016-4-25
add --------
听老土讲徽州14讲
http://www.517huizhou.com/forum.php?mod=viewthread&tid=91007&extra=page%3D1
2016-4-29
add --------
图说徽州系列讲座第十二讲承志堂的元宝梁
http://www.517huizhou.com/forum.php?mod=viewthread&tid=90726&extra=page%3D1
2016-4-12
add --------
看图说徽州第九讲明代正德十一年砖雕
http://www.517huizhou.com/forum.php?mod=viewthread&tid=90361&extra=page%3D1
2016-3-25
add --------
宏村承志堂的宣传是对汪家人的摸黑，强烈要求把排山阁，吸鸦片烟的吞云轩的匾额拆除
http://www.517huizhou.com/forum.php?mod=viewthread&tid=90527&extra=page%3D1
2016-4-1
add --------
答题赠书
http://www.517huizhou.com/forum.php?mod=viewthread&tid=90423&extra=page%3D1
2016-3-28
add --------
看图说徽州第8讲
http://www.517huizhou.com/forum.php?mod=viewthread&tid=90193&extra=page%3D1
2016-3-17
add --------
看图说徽州第六讲隔扇门
http://www.517huizhou.com/forum.php?mod=viewt

In [21]:
print idx

520


In [69]:
print item

<tbody id="normalthread_74385">
<tr>
<td class="icn">
<a href="forum.php?mod=viewthread&amp;tid=74385&amp;extra=page%3D4" target="_blank" title="新窗口打开">
<img src="static/image/common/folder_common.gif"/>
</a>
</td>
<th class="common">
<a class="showcontent y" href="javascript:;" id="content_74385" onclick="CONTENT_TID='74385';CONTENT_ID='normalthread_74385';showMenu({'ctrlid':this.id,'menuid':'content_menu'})" title="更多操作"></a>
<a class="tdpre y" href="javascript:void(0);" onclick="previewThread('74385', 'normalthread_74385');">预览</a>
 <a class="s xst" href="forum.php?mod=viewthread&amp;tid=74385&amp;extra=page%3D4" onclick="atarget(this)">绩溪紫园</a>
</th>
<td class="by">
<cite>
<a c="1" href="home.php?mod=space&amp;uid=25800" mid="card_7235" style="color: #559955;">清风溪水。</a></cite>
<em><span>2014-8-11</span></em>
</td>
<td class="num"><a class="xi2" href="forum.php?mod=viewthread&amp;tid=74385&amp;extra=page%3D4">3</a><em>525</em></td>
<td class="by">
<cite><a c="1" href="home.php?mod=s

In [153]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4496 entries, 0 to 4495
Data columns (total 11 columns):
dt          4496 non-null object
source      4496 non-null object
domain      4496 non-null object
url         4496 non-null object
title       4496 non-null object
summary     4496 non-null object
image       4496 non-null object
brief       4496 non-null object
body        4496 non-null object
raw_page    4496 non-null object
from        4496 non-null object
dtypes: object(11)
memory usage: 421.5+ KB


In [148]:
df.head()

dt source domain                                                url  \
0       4 天前  徽州大论坛         http://www.517huizhou.com/forum.php?mod=viewth...   
1  2017-2-25  徽州大论坛         http://www.517huizhou.com/forum.php?mod=viewth...   
2  2017-2-23  徽州大论坛         http://www.517huizhou.com/forum.php?mod=viewth...   
3  2017-2-13  徽州大论坛         http://www.517huizhou.com/forum.php?mod=viewth...   
4  2017-2-15  徽州大论坛         http://www.517huizhou.com/forum.php?mod=viewth...   

                              title summary image brief body raw_page from  
0             方利山：国宝级文物— 汪华墓“唐宋诰敕”碑                                         
1                  方利山文：为朱熹《朱子家训》正名                                         
2           菲元：善用黄山“黄帝文化”资源，加劲打好黄山牌                                         
3                            盛世修谱~~                                         
4  方利山：承续文脉 保护徽州文化生态——贺箬岭古道汪王总庙修复落成

In [26]:
if(len(df) != 0):
    df.to_csv('../data/topics/400/517huizhou.csv', encoding='utf-8', index=False)

In [158]:
idx = 4462

In [161]:
for i in range(4462, len(df)):
    url = df.loc[i]['url']
    browser.get(url)
    time.sleep(15)
    
    soup = BeautifulSoup(browser.page_source)
    sub_text = soup.findAll('div', {'class':'t_fsz'})
    
    fst = 0
    for item in sub_text:
        if fst == 0:
            #print item.text
            print i, df.loc[i]['title']
            #print 'add --', item.text
            df.loc[i]['body'] = item.text
            df.loc[i]['raw_page'] = browser.page_source    
            fst = 1


4462 [原创]期盼，生活的至美
4463 笑 的好处
4464 论坛情（打油诗）
4465 [原创]七律。汪口有感
4466 [原创]徽州多有奇女子
4467 [原创]五律。静坐养生莫迟疑！
4468 [原创]题齐云山云岩湖
4469 [原创]七律。酒品差，没办法！
4470 不一样的夏天，朋友你的感觉和我一样吗？
4471 随笔
4472 [原创]七律。自辱
4473 有些事，从心里泛起！
4474 [原创]五律。夏夜乘凉
4475 游陶辛！
4476 很多事，让我忆起……
4477 [原创]进县见友人空调有感
4478 《天香》凤尾猴魁咏
4479 漂浮
4480 荷塘月色
4481 黄山咏
4482 [原创]乡村邻里情意醇
4483 慧心读网
4484 [原创]七律。忆水碓
4485 菜场！
4486 旋转木马
4487 活着谁看
4488 [原创]七律。凤山三宝之——西门井
4489 [原创]“凤山三宝之——鱼塘”有感
4490 [原创]七绝。欣闻安徽电视台来浙岭
4491 [原创]七律。“小桥流水人家——李坑”印象
4492 [原创]凤山八景之——花桥头
4493 淡淡的老冰棍
4494 闻香寻兰
4495 午夜思绪


In [107]:
print url

forum.php?mod=viewthread&tid=195025&extra=page%3D1%26filter%3Dtypeid%26typeid%3D162


In [162]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4496 entries, 0 to 4495
Data columns (total 11 columns):
dt          4496 non-null object
source      4496 non-null object
domain      4496 non-null object
url         4496 non-null object
title       4496 non-null object
summary     4496 non-null object
image       4496 non-null object
brief       4496 non-null object
body        4496 non-null object
raw_page    4496 non-null object
from        4496 non-null object
dtypes: object(11)
memory usage: 421.5+ KB


In [12]:
for i in range(0, 1):
    url_0 = news_url
    print url_0
    browser.get(url_0)
    time.sleep(5)

    soup = BeautifulSoup(browser.page_source)
    sub_text = soup.findAll('div', {'class':'listA_title'})

    for item in sub_text:
        t1 = item.find('div', {'class':'listA_Time'})
        #print 'www.hzwh.com/' + item.a['href']
        #print t1.text
        print item.a.text
        url = 'www.hzwh.com/' + item.a['href']
        
        not_existed = True
        #check if in cleaned set
        if t1 != None and len(df_check) != 0:
            t0 = abs(sum(df_check['url'].str.find(url))) 
            if t0 != len(df_check):
                not_existed = False

        #check if in dynamic set
        if t1 != None and len(df) != 0 and not_existed == True:
            t0 = abs(sum(df['url'].str.find(str(url)))) 
            if t0 != len(df):
                not_existed = False

        #print t0, len(df), t1

        if not_existed == True and t1 != None:   
            print "-------- add"
            df.loc[idx] = get_empty_columns()        
            df.loc[idx]['url'] = url
            df.loc[idx]['title'] = item.a.text
            df.loc[idx]['source'] = '徽州文化网'
            df.loc[idx]['dt'] = t1.text
            idx += 1

http://www.hzwh.com/toplistb.asp?id=229&page=0
传统徽州仁商义贾践行商君子之道的历史启示
《阅读徽州》第七编“徽州科技”
传统徽商的诚信为本和仁义向善
传统徽商的诚信为本和仁义向善
黟县古城改造重要的是务须保住文教根脉
推动徽州传统文化的普及宣传
徽学与当代学术 
“儒学与地域文化：徽学国际学术研讨会”综述
洪经纶墓葬地点的疏证
徽州乡镇志中所见明清民间救荒措施
徽州文化全书
论文化生产力与中部地区的经济发展
徽州书院的讲会
明代海上徽帮与宁波海上丝绸之路
徽州昌溪周氏家族与宁波近代商业发展史
徽商历史的文化价值批判 
徽商
徽商文化及特点
婺源徽剧音乐研究
古徽州和谐乡村社会建设的经验


/usr/local/lib/python2.7/site-packages/bs4/__init__.py:181: UserWarning: No parser was explicitly specified, so I'm using the best available HTML parser for this system ("html5lib"). This usually isn't a problem, but if you run this code on another system, or in a different virtual environment, it may use a different parser and behave differently.

The code that caused this warning is on line 162 of the file /usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py. To get rid of this warning, change code that looks like this:

 BeautifulSoup([your markup])

to this:

 BeautifulSoup([your markup], "html5lib")

  markup_type=markup_type))


In [13]:
print len(df)

0


In [12]:
df.head(50)

dt source domain                              url  \
0   2004-03-03  徽州文化网         www.hzwh.com/article.asp?id=533   
1   2004-03-03  徽州文化网         www.hzwh.com/article.asp?id=532   
2   2004-03-09  徽州文化网         www.hzwh.com/article.asp?id=530   
3   2004-03-22  徽州文化网         www.hzwh.com/article.asp?id=528   
4   2004-03-31  徽州文化网         www.hzwh.com/article.asp?id=527   
5   2004-04-02  徽州文化网         www.hzwh.com/article.asp?id=525   
6   2004-04-11  徽州文化网         www.hzwh.com/article.asp?id=523   
7   2004-04-12  徽州文化网         www.hzwh.com/article.asp?id=521   
8   2004-04-15  徽州文化网         www.hzwh.com/article.asp?id=520   
9   2004-04-15  徽州文化网         www.hzwh.com/article.asp?id=518   
10  2004-05-11  徽州文化网         www.hzwh.com/article.asp?id=517   
11  2004-05-13  徽州文化网         www.hzwh.com/article.asp?id=516   
12  2009-04-28  徽州文化网         www.hzwh.com/article.asp?id=505   

                  title summary image brief body raw_page from  
0            旧时徽州税票名称种种                                         
1           小小大阜村 爷孙宰相俩                                         
2            李良春与《惯痴文集》                                         
3                 古老的石林                                         
4                 洪钧行书联                                         
5            万安“中国罗盘之乡”                                         
6                   吕仙宫                                         
7               溪 头 探 古                                         
8        五绝《小桃源》中否李白所作？                                         
9               解读戴震诗一首                                         
10        安徽休宁——中国第一状元县                                         
11           独树一帜的徽州税文化                                         
12  扬弃古徽州廉政文化 促进新黄山廉政建设

In [28]:
print len(df)

0


In [163]:
def url_get(url):
    browser.get(url)
    print url
    time.sleep(5)
    return browser.page_source

In [164]:
df['dt'] = df_check['dt']
df['source'] = df_check['source']
df['url'] = df_check['url']
df['title'] = df_check['title']
df['raw_page'] = df_check.url.map(lambda x: url_get('http://' + x) )  

http://www.hzwh.com/article.asp?id=2538
http://www.hzwh.com/article.asp?id=2537
http://www.hzwh.com/article.asp?id=2536
http://www.hzwh.com/article.asp?id=2535
http://www.hzwh.com/article.asp?id=2534
http://www.hzwh.com/article.asp?id=2533
http://www.hzwh.com/article.asp?id=2099
http://www.hzwh.com/article.asp?id=2098
http://www.hzwh.com/article.asp?id=1848
http://www.hzwh.com/article.asp?id=1847
http://www.hzwh.com/article.asp?id=1846
http://www.hzwh.com/article.asp?id=1794
http://www.hzwh.com/article.asp?id=1695
http://www.hzwh.com/article.asp?id=1694
http://www.hzwh.com/article.asp?id=1693
http://www.hzwh.com/article.asp?id=1624
http://www.hzwh.com/article.asp?id=1623
http://www.hzwh.com/article.asp?id=1622
http://www.hzwh.com/article.asp?id=1455
http://www.hzwh.com/article.asp?id=1418
http://www.hzwh.com/article.asp?id=1417
http://www.hzwh.com/article.asp?id=1416
http://www.hzwh.com/article.asp?id=1415
http://www.hzwh.com/article.asp?id=1414
http://www.hzwh.com/article.asp?id=1347


TimeoutException: Message: Error loading page, timed out (onDOMContentLoaded)


In [165]:
df['url'] = 'http://' + df['url']

In [119]:
df.head()

dt source domain  \
0    2017-2-9   故园徽州          
1    2017-1-1   故园徽州          
2  2016-11-23   故园徽州          
3  2016-10-25   故园徽州          
4  2016-10-17   故园徽州          

                                                 url  \
0  forum.php?mod=viewthread&tid=195025&extra=page...   
1  forum.php?mod=viewthread&tid=194254&extra=page...   
2  forum.php?mod=viewthread&tid=192264&extra=page...   
3  forum.php?mod=viewthread&tid=189658&extra=page...   
4  forum.php?mod=viewthread&tid=189555&extra=page...   

                        title summary image brief  \
0  【春节民俗活动预告】——中云镇桃溪坑头“抬汪公”民俗                       
1               2017.01.01的早上                       
2                       上学去喽！                       
3        流淌的登源河（29）：老造反成了普通看客                       
4                    沙溪“淩”姓来由                       

                                                body  \
0  \n\n\n\n\n马上注册成为会员，享用更多功能，让你轻松玩转徽州\n您需要 登录 才可以...   
1  \n\n\n\n\n马上注册成为会员，享用更多功能，让你轻松玩转徽州\n您需要 登录 才可以...   
2  \n\n\n\n\n马上注册成为会员，享用更多功能，让你轻松玩转徽州\n您需要 登录 才可以...   
3  \n\n\n\n\n马上注册成为会员，享用更多功能，让你轻松玩转徽州\n您需要 登录 才可以...   
4  \n\n\n\n\n马上注册成为会员，享用更多功能，让你轻松玩转徽州\n您需要 登录 才可以...   

                                            raw_page from  
0  <html xmlns="http://www.w3.org/1999/xhtml"><he...       
1  <html xmlns="http://www.w3.org/1999/xhtml"><he...       
2  <html xmlns="http://www.w3.org/1999/xhtml"><he...       
3  <html xmlns="http://www.w3.org/1999/xhtml"><he...       
4  <html xmlns="http://www.w3.org/1999/xhtml"><he...

In [152]:
characters_num = 40
def get_brief(txt):
    soup = BeautifulSoup(txt)
    sub_text = soup.findAll('div', {'class':'article_contents'})
    for item in sub_text:
        t20 = item.text

    idx = 2
    t30 = t20.split(u'\n')
    
    cnt = 0
    tt = ''
    idx = 0
    for k in range(0, len(t30)):
        if (len(t30[k]) > 40 and len(tt) < 40):
            idx = k
            tt = t30[k].strip()
    
    if len(tt) == 0:
        tt = t20.strip()
                
    #print idx

    t40 = tt[0:characters_num]
    t50 = t40.encode('utf-8') + '..'
    #print idx, t40
    return t50

In [158]:
df['brief'] = df.raw_page.map(lambda x: get_brief(x) )  

In [154]:
def get_body(txt):
    soup = BeautifulSoup(txt)
    sub_text = soup.findAll('div', {'class':'article_contents'})
    for item in sub_text:
        t20 = item.text

    return t20

In [159]:
df['body'] = df.raw_page.map(lambda x: get_body(x) )  

cnt = 0
for item in df['brief']:
    print cnt, item
    cnt += 1

cnt = 0
for item in df['body']:
    print cnt, item
    cnt += 1

soup = BeautifulSoup(df.loc[527]['raw_page'])
sub_text = soup.findAll('div', {'class':'article_contents'})
for item in sub_text:
    t20 = item.text
t30 = t20.split(u'\n')

tt = t20.strip()
print len(t30), tt[0:40], t20

In [161]:
df.head()

dt source domain                               url  \
0  2016-04-08  徽州文化网    NaN  www.hzwh.com/article.asp?id=2538   
1  2016-04-08  徽州文化网    NaN  www.hzwh.com/article.asp?id=2537   
2  2016-04-08  徽州文化网    NaN  www.hzwh.com/article.asp?id=2536   
3  2016-04-08  徽州文化网    NaN  www.hzwh.com/article.asp?id=2535   
4  2016-04-08  徽州文化网    NaN  www.hzwh.com/article.asp?id=2534   

                  title summary image  \
0  传统徽州仁商义贾践行商君子之道的历史启示     NaN   NaN   
1       《阅读徽州》第七编“徽州科技”     NaN   NaN   
2        传统徽商的诚信为本和仁义向善     NaN   NaN   
3        传统徽商的诚信为本和仁义向善     NaN   NaN   
4    黟县古城改造重要的是务须保住文教根脉     NaN   NaN   

                                        brief  \
0  全国人大代表、民盟安徽省委副主委、安徽省文联副主席钱念孙先生在《光明日报》201..   
1  长期以来，得益于灵山秀水自然环境的滋养、兴学重教社会风气的熏染、经世致用治学风格..   
2  徽州古楹联云“仁中取利真君子/义内求财大丈夫”。这是对传统徽商群体中那些仁商义贾..   
3  徽州古楹联云“仁中取利真君子/义内求财大丈夫”。这是对传统徽商群体中那些仁商义贾..   
4  黟县进行古城改造并申报国家级历史文化名城总体方案正在逐步完善，各项工作正在有条不..   

                                                body  \
0  传统徽州仁商义贾践行商君子之道的历史启示\n郗延红\n全国人大代表、民盟安徽省委副主委、安徽...   
1  《阅读徽州》第七编“徽州科技”（市委市政府通过市社科联组织编写）\n中国科学技术学大出版社出...   
2  传统徽商的诚信为本和仁义向善\n\n郗延红\n徽州古楹联云“仁中取利真君子/义内求财大丈夫”...   
3  传统徽商的诚信为本和仁义向善\n \n郗延红\n徽州古楹联云“仁中取利真君子/义内求财大丈夫...   
4  黟县古城改造重要的是务须保住文教根脉\n                         郗...   

                                            raw_page from  
0  <html xmlns="http://www.w3.org/1999/xhtml"><he...  NaN  
1  <html xmlns="http://www.w3.org/1999/xhtml"><he...  NaN  
2  <html xmlns="http://www.w3.org/1999/xhtml"><he...  NaN  
3  <html xmlns="http://www.w3.org/1999/xhtml"><he...  NaN  
4  <html xmlns="http://www.w3.org/1999/xhtml"><he...  NaN

In [127]:
df.loc[24]['url']

'www.hzwh.com/article.asp?id=1347'

In [121]:
if(len(df) != 0):
    df.to_csv('../data/complete/hzwh_huixueyuanpu.csv', encoding='utf-8', index=False)

In [27]:
if(browser):
    browser.quit()

In [31]:
for item in df['title']:
    print item

In [32]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 11 columns):
dt          0 non-null object
source      0 non-null object
domain      0 non-null object
url         0 non-null object
title       0 non-null object
summary     0 non-null object
image       0 non-null object
brief       0 non-null object
body        0 non-null object
raw_page    0 non-null object
from        0 non-null object
dtypes: object(11)
memory usage: 0.0+ bytes


In [173]:
df2 = pd.read_csv('../data/complete/hzwh_huixueyuanpu.csv')

In [174]:
df2.head()

dt source  domain                                      url  \
0  2016-04-08  徽州文化网     NaN  http://www.hzwh.com/article.asp?id=2538   
1  2016-04-08  徽州文化网     NaN  http://www.hzwh.com/article.asp?id=2537   
2  2016-04-08  徽州文化网     NaN  http://www.hzwh.com/article.asp?id=2536   
3  2016-04-08  徽州文化网     NaN  http://www.hzwh.com/article.asp?id=2535   
4  2016-04-08  徽州文化网     NaN  http://www.hzwh.com/article.asp?id=2534   

                  title  summary  image  \
0  传统徽州仁商义贾践行商君子之道的历史启示      NaN    NaN   
1       《阅读徽州》第七编“徽州科技”      NaN    NaN   
2        传统徽商的诚信为本和仁义向善      NaN    NaN   
3        传统徽商的诚信为本和仁义向善      NaN    NaN   
4    黟县古城改造重要的是务须保住文教根脉      NaN    NaN   

                                        brief  \
0  全国人大代表、民盟安徽省委副主委、安徽省文联副主席钱念孙先生在《光明日报》201..   
1  长期以来，得益于灵山秀水自然环境的滋养、兴学重教社会风气的熏染、经世致用治学风格..   
2  徽州古楹联云“仁中取利真君子/义内求财大丈夫”。这是对传统徽商群体中那些仁商义贾..   
3  徽州古楹联云“仁中取利真君子/义内求财大丈夫”。这是对传统徽商群体中那些仁商义贾..   
4  黟县进行古城改造并申报国家级历史文化名城总体方案正在逐步完善，各项工作正在有条不..   

                                                body  \
0  传统徽州仁商义贾践行商君子之道的历史启示\n郗延红\n全国人大代表、民盟安徽省委副主委、安徽...   
1  《阅读徽州》第七编“徽州科技”（市委市政府通过市社科联组织编写）\n中国科学技术学大出版社出...   
2  传统徽商的诚信为本和仁义向善\n\n郗延红\n徽州古楹联云“仁中取利真君子/义内求财大丈夫”...   
3  传统徽商的诚信为本和仁义向善\n \n郗延红\n徽州古楹联云“仁中取利真君子/义内求财大丈夫...   
4  黟县古城改造重要的是务须保住文教根脉\n                         郗...   

                                            raw_page  from  
0  <html xmlns="http://www.w3.org/1999/xhtml"><he...   NaN  
1  <html xmlns="http://www.w3.org/1999/xhtml"><he...   NaN  
2  <html xmlns="http://www.w3.org/1999/xhtml"><he...   NaN  
3  <html xmlns="http://www.w3.org/1999/xhtml"><he...   NaN  
4  <html xmlns="http://www.w3.org/1999/xhtml"><he...   NaN